### DISCUSSION TITLE
COVID Vaccines - How to identify Shielding/CEV and At Risk/CV groups

### DISCUSSION THREAD

**Discussion by HelenCEBM** [opensafely/documentation#486] [Show and tell]
# Background

There were two groups of clinically vulnerable people in relation to COVID and the vaccination programme. 

## 1. "Shielding"
AKA **Clinically Extremely Vulnerable (CEV)** people who were advised to shield to avoid exposure to the virus.
- A shielding (["high risk from COVID-19"](https://www.opencodelists.org/codelist/primis-covid19-vacc-uptake/shield/v1/)) flag was added to their GP record through various possible routes, for example through the initial nationally applied algorithm (as far as I know the details are not available), by their GP or hospital doctor, or later by the Q-COVID algorithm (February 2021).  
- Patients may also have been removed by various routes, e.g. through their condition improving. These patients have a ["low risk" or "medium risk" flag](https://www.opencodelists.org/codelist/primis-covid19-vacc-uptake/nonshield/v1/) added to their GP record to indicate they were no longer considered "high risk". 
- The CEV group is therefore relatively simply defined as: having a flag for being at [increased risk from COVID](https://www.opencodelists.org/codelist/primis-covid19-vacc-uptake/shield/v1/) and NOT having a _more recent_ flag for being at [medium or low risk](https://www.opencodelists.org/codelist/primis-covid19-vacc-uptake/nonshield/v1/). 
- For **vaccination** purposes, this group were in group 4 of the 9 initial vaccine priority groups, and covered ages 16-69 (many older people 70+ were also shielding, but they had already become eligible for vaccination due to their age). 

## 2. "At Risk"
AKA **Clinically Vulnerable (CV)** people were not advised to shield but may have been at increased risk from COVID-19 compared to others of the same age due to less severe conditions. 
- The CV/at-risk group is much more complex because it looks at various conditions, taking into account prescriptions, hospitalisations, diagnoses and "resolved" codes for each one.
    - e.g. patients are excluded from the diabetes group if they have a "diabetes resolved" code more recently than a diabetes code; 
    - asthmatics may have certain diagnosis codes, a series of prescriptions OR hospitalisation; etc.  
- For **vaccination** purposes, this group were in group 6 of the 9 initial vaccine priority groups. This covered ages 16-64 (many older people 65+ were also At Risk, but they had already become eligible for vaccination due to their age). 

# Identifying Shielding/CEV And CV/At Risk Groups in Opensafely

## Codelists  
- Codelists from the official specification are here https://www.opencodelists.org/codelist/primis-covid19-vacc-uptake/
- Note these codelists are correct as of ~January 2021, we have not included the various more recent versions which have since been released. 

## Example Repositories
We have implemented the codelists using a few slightly different structures in different repos, though they should ultimately achieve the same results. 

### Shielding / CEV 
- For our [weekly report](https://reports.opensafely.org/reports/vaccine-coverage) we create the CEV (shielding) flag entirely within the study definition, comparing the latest date of any high risk and lower risk flags for each patient ([link](https://github.com/opensafely/nhs-covid-vaccination-coverage/blob/588c2d53478d59724b42609a9b0074ab9fa6cf16/analysis/study_definition_delivery_common.py#L268-L294)). 
- For our paper which combined TPP and EMIS we only extract the dates of the "high risk" and "lower risk" records in the [study definition](https://github.com/opensafely/covid19-vaccine-coverage-tpp-emis/blob/79195309b06608baffb0d3eb57f7b99bfec17fcc/analysis/study_definition.py#L262-L276) and then use separate Python scripts to compare the dates ([here](https://github.com/opensafely/covid19-vaccine-coverage-tpp-emis/blob/70b041d295c4ccdb80f23009206d080a7dab74b4/analysis/add_groupings.py#L143-L151) and [here](https://github.com/opensafely/covid19-vaccine-coverage-tpp-emis/blob/70b041d295c4ccdb80f23009206d080a7dab74b4/analysis/transform_fast.py#L189-L192)). This may be quite a complex set of scripts to follow but the reason for this structure was that we auto-generated the scripts from the specification document to reduce manual labour. 

### CV/At risk
- We have not implemented this group in our weekly report. 
- For our [paper](https://bjgp.org/content/early/2021/09/24/BJGP.2021.0376), as for CEV, we only extract the date of each individual event (e.g. prescriptions, diagnoses) in the [study definition](https://github.com/opensafely/covid19-vaccine-coverage-tpp-emis/blob/79195309b06608baffb0d3eb57f7b99bfec17fcc/analysis/study_definition.py#L262-L276) and then use separate Python scripts to implement the logic for each condition in turn and then make a combined at-risk group ([here](https://github.com/opensafely/covid19-vaccine-coverage-tpp-emis/blob/70b041d295c4ccdb80f23009206d080a7dab74b4/analysis/add_groupings.py#L5-L84) and [here](https://github.com/opensafely/covid19-vaccine-coverage-tpp-emis/blob/70b041d295c4ccdb80f23009206d080a7dab74b4/analysis/transform_fast.py#L173-L209)).
- In some ongoing work we have instead implemented most of the date comparisons in the [study definition](https://github.com/opensafely/post-vax-ae-attendance/blob/bd708e29bfa3d7162ffe32fc08e3146666fa4bfb/analysis/study_definition.py#L668-L685) then made the combined CV/at risk group [in R](https://github.com/opensafely/post-vax-ae-attendance/blob/main/analysis/process.R); this was done this way to fit it in around the existing analysis structure.

**Comment by elsie-h** [comment]
Thanks @HelenCEBM - this is really helpful. 

I've got a question about defining JCVI group 1 (older adults resident in care homes). Is this [primis code list](https://www.opencodelists.org/codelist/primis-covid19-vacc-uptake/longres/v1/) considered sufficient for defining that group, or should any additional conditions on age be added? For example, I wondered if the code list alone might identify younger people with learning disabilities who are in long-term residential care?

**Reply by HelenCEBM** [reply]
Yes you will identify some younger people (but not all, I think) so I would say it's a good idea to limit to 65+. We did this in our vaccine-declined report https://www.medrxiv.org/content/10.1101/2021.08.05.21259863v1.full-text (see "Priority groups for vaccination" and Box 1).

It's also useful to limit the pregnancy `preg` / `pregdel` flags to females of childbearing age, if you're using those.
